### PR TITLE
Basic Camera Control

### PR DESCRIPTION
**Pull request by dogmaphobic**
This implements the __*Basic Camera Operations*__ described by the [Camera Protocol Documentation](https://mavlink.io/en/protocol/camera.html).

Only the very basic commands are implemented, providing image capture only (no video capture). Within the basic set of messages, all messages are implemented as well as all behaviors.

This "camera" component handles its own UDP "link" with the GCS. It emits its own heartbeats, and handles all the camera related communication between the GCS and the camera.

Next will be implementing some parameters and options for controlling the "camera".

### CODE REVIEW

File: src/gazebo_geotagged_images_plugin.cpp [hunk 1]
```diff
@@ -31,346 +31,585 @@ using namespace std;
...
-  if (len <= 0) {
-    printf("Failed sending mavlink message\n");
-  }
+void GeotaggedImagesPlugin::_handle_storage_info(const mavlink_message_t *pMsg, struct sockaddr* srcaddr)
+{
+    gzdbg << "Send storage info" << endl;
+    float total_mib     = 0.0f;
+    float available_mib = 0.0f;
+    boost::filesystem::space_info si = boost::filesystem::space(".");
+    available_mib = (float)((double)si.available / (1024.0 * 1024.0));
```
Comment (julianoes) [inline src/gazebo_geotagged_images_plugin.cpp:596]: Haha, that's usually plenty 😄

File: src/gazebo_geotagged_images_plugin.cpp [hunk 2]
```diff
@@ -31,346 +31,585 @@ using namespace std;
...
+    mavlink_message_t msg;
+    mavlink_msg_camera_capture_status_pack_chan(
+        1,
+        MAV_COMP_ID_CAMERA,
+        MAVLINK_COMM_1,
+        &msg,
+        0,
+        status,                                 // image status
+        0,                                      // video status (Idle)
+        _captureMode == CAPTURE_ELAPSED ? (float)_captureInterval : 0, // image interval
```
Comment (julianoes) [inline src/gazebo_geotagged_images_plugin.cpp]: `_captureMode` is not guarded by `_captureMutex` here. Is this on purpose?

**Review by julianoes — APPROVED** [review]
Reviewed, only one small lock question, otherwise ok.

File: src/gazebo_geotagged_images_plugin.cpp (hunk 2 above)
Comment (dogmaphobic) [reply src/gazebo_geotagged_images_plugin.cpp]: Nope, I missed it.

**Pull request merged**